src/browser/api/external_application.ts

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2017 OpenFin Inc.
+
+Licensed under OpenFin Commercial License you may not use this file except in compliance with your Commercial License.
+Please contact OpenFin Inc. at sales@openfin.co to obtain a Commercial License.
+*/
+import ofEvents from '../of_events';
+
+import { Identity } from '../../shapes';
+
+import * as ProcessTracker from '../process_tracker.js';
+
+const authenticatedConnections: any[] = [];
+const connectedEvent = 'external-application/connected';
+const disconnectedEvent = 'external-application/disconnected';
+
+export module ExternalApplication {
+    export function addEventListener(identity: Identity, type: string, listener: Function) {
+        const evt = `external-application/${type}/${identity.uuid}`;
+        ofEvents.on(evt, listener);
+
+        return () => {
+            ofEvents.removeListener(evt, listener);
+        };
+    }
+
+    export function removeEventListener(identity: Identity, type: string, listener: Function) {
+        ofEvents.removeListener(`external-application/${type}/${identity.uuid}`, listener);
+    }
+
+    export function getInfo(externalApp: Identity): ExternalProcessInfo {
+        const process: any = ProcessTracker.getProcessByUuid(externalApp.uuid);
```

Comment:
`process` is already a [node global](https://nodejs.org/api/process.html), perhaps let's not shadow that?